org.eclipse.scanning.example/src/org/eclipse/scanning/example/detector/RandomLineDevice.java

```diff
@@ -103,10 +124,44 @@ public boolean write(IPosition pos) throws ScanningException {
 		return true;
 	}
 
-	@Override
-	public void configure(RandomLineModel model) throws ScanningException {	
-		super.configure(model);
-		setName(model.getName());
+	protected void count(StackTraceElement[] ste) {
+		String methodName = getMethodName(ste);
+		Integer count = counts.get(methodName);
+		if (count==null) count = 0;
+		count = count+1;
+		counts.put(methodName, count);
+	}
+	
+	public int getCount(String method) {
+		if (!counts.containsKey(method)) return 0;
+		return counts.get(method);
+	}
+	
+	protected static final String getMethodName ( StackTraceElement ste[] ) {  
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the array designator from the variable to the type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1197)